src/main/scala/com/microsoft/hyperspace/index/IndexLogEntryTags.scala

```diff
@@ -16,12 +16,52 @@
 
 package com.microsoft.hyperspace.index
 
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
+
+import com.microsoft.hyperspace.util.HyperspaceConf
+
 object IndexLogEntryTags {
-  // HYBRIDSCAN_REQUIRED indicates if Hybrid Scan is required for this index or not.
+  // HYBRIDSCAN_REQUIRED indicates if Hybrid Scan is required for the index or not.
   val HYBRIDSCAN_REQUIRED: IndexLogEntryTag[Boolean] =
     IndexLogEntryTag[Boolean]("hybridScanRequired")
 
   // COMMON_SOURCE_SIZE_IN_BYTES stores overlapping bytes of index source files and given relation.
   val COMMON_SOURCE_SIZE_IN_BYTES: IndexLogEntryTag[Long] =
     IndexLogEntryTag[Long]("commonSourceSizeInBytes")
+
+  // IS_SIGNATURE_MATCH indicates if the plan has the same signature value with the index or not.
+  val IS_SIGNATURE_MATCH: IndexLogEntryTag[Boolean] =
+    IndexLogEntryTag[Boolean]("isSignatureMatch")
+
+  // IS_HYBRIDSCAN_CANDIDATE indicates if the index can be applied to the plan using Hybrid Scan.
+  val IS_HYBRIDSCAN_CANDIDATE: IndexLogEntryTag[Boolean] =
+    IndexLogEntryTag[Boolean]("hybridScanCandidate")
+
+  // IS_HYBRIDSCAN_CONFIG_CAPTURE contains values of Hybrid Scan related configs.
+  val HYBRIDSCAN_CONFIG_CAPTURE: IndexLogEntryTag[Seq[String]] =
+    IndexLogEntryTag[Seq[String]]("hybridScanConfigCapture")
+
+  private def getHybridScanConfigs(spark: SparkSession): Seq[String] = {
+    Seq(
+      HyperspaceConf.hybridScanEnabled(spark).toString,
+      HyperspaceConf.hybridScanDeletedRatioThreshold(spark).toString,
+      HyperspaceConf.hybridScanAppendedRatioThreshold(spark).toString)
+  }
+
+  private[hyperspace] def resetHybridScanTagsIfNeeded(
+      spark: SparkSession,
+      plan: LogicalPlan,
+      indexes: Seq[IndexLogEntry]): Unit = {
+    val curConfigs = getHybridScanConfigs(spark)
+
+    indexes.foreach { index =>
+      val taggedConfigs = index.getTagValue(plan, HYBRIDSCAN_CONFIG_CAPTURE)
+      if (taggedConfigs.isEmpty || !taggedConfigs.get.equals(curConfigs)) {
```

Comment:
nit: prefer using `map` or `foreach` for this pattern `taggedConfigs.isEmpty || taggedConfigs.get`:
```
      index.getTagValue(plan, HYBRIDSCAN_CONFIG_CAPTURE).foreach { taggedConfigs =>
        if (taggedConfigs.equals(curConfigs)) {
          // Need to reset cached tags as these config changes can change the result.
          index.unsetTagValue(plan, IS_HYBRIDSCAN_CANDIDATE)
          index.setTagValue(plan, HYBRIDSCAN_CONFIG_CAPTURE, curConfigs)
        }
      }
```